### ISSUE TITLE
eBPF + L2Announce + without-kube-proxy + destination pods on separate k8s node from in-bound traffic = eventual guaranteed breaking

### ISSUE THREAD

**Issue by BloodyIron** [cilium/cilium#33295]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When I have my destination pods on a k8s node separate from the L2Announce leader k8s node, after a seemingly random period of time, those pods will eventually be 100% unreachable and not self-heal.

### Cilium Version

v1.15.1

### Kernel Version

5.15.0-112-generic #122-Ubuntu SMP

### Kubernetes Version

v1.26.15+rke2r1

### Regression

I don't know yet

### Sysdump

NA

### Relevant log output

```shell
I can't tell what log output actually is relevant, despite checking logs that align time-wise. Nothing seems consistent after multiple break iterations!
```

### Anything else?

I am using:

1. Rancher
2. RKE2 for nodes
3. ArgoCD for greatness
4. Proxmox VE for running the VMs that are k8s nodes
5. Ubuntu 22.04 for the VMs

In the relevant test cluster I have 3x RKE2 nodes, and I'm using the RKE2 documented method to disable kube-proxy. Then in-turn have cilium do Layer 2 ARP broadcasting and LoadBalancing for a single LAN IP outside the cluster. I do not want to use BGP that is not okay in this environment.

In my test scenario, I have two relevant pods, NGINX Ingress, and librespeed. librespeed goes through NGINX Ingress, and NGINX Ingress has an 80/443 Service object.

The two pods are on the same node as each other, but a different node from the ARP IP broadcast node (leader).

Traffic comes in, I get correct Source IP, that's desirable.

However the following problems occur:
1. If the two pods get terminated and regenerate, then they are 100% unreachable, every time. This does not self-heal and requires all Cilium pods be terminated and re-init'd, and/or redeployed.
2. Eventually after a random period of time between 3 minutes and 1h 30min (ish) the NGINX Ingress Service (or something like that) will just suddenly stop getting traffic (I'm reloading the librespeed page once per minute to validate this). And this never self-heals, and requires all Cilium pods be terminated and re-init'd, and/or redeployed.

For Rancher's RKE2 stuff I have the following contents for the rke2-cilium-config.yaml file, and a daemonset that distributes this content (including updates) to all nodes in the cluster:

```
  rke2-cilium-config.yaml: |
    apiVersion: helm.cattle.io/v1
    kind: HelmChartConfig
    metadata:
      name: rke2-cilium
      namespace: kube-system
    spec:
      valuesContent: |-
        rollOutCiliumPods: true
        kubeProxyReplacement: true
        kubeProxyReplacementHealthzBindAddr: 0.0.0.0:10256
        k8sServiceHost: 127.0.0.1
        k8sServicePort: 6443
        k8sClientRateLimit:
          qps: 20
          burst: 30
        l2announcements:
          enabled: true
          leaseDuration: 5s
          leaseRenewDeadline: 3s
          leaseRetryPeriod: 200ms
        routingMode: tunnel
        tunnelProtocol: geneve
        loadBalancer:
          mode: dsr
          dsrDispatch: geneve
        socketLB:
          hostNamespaceOnly: true
        debug:
          enabled: true
          verbose: flow
```

Changes to this file triggers re-deploy for Cilium via Helm Chart, so these generally represent launch parameters for Cilium itself.

For the CiliumL2AnnouncementPolicy I only really have "loadBalancerIPs: true" defined as I have not seen any reason to enable anything else to improve this situation, and have been trying to limit my variables in this scenario.

I'm at my wit's end, I cannot find why this is failing, even after turning debug logging on (as you see).

I've followed both RKE2 and Cilium's documentation on getting this going, and I can't for the life of me tell what I'm doing wrong. Please help, I need this working and I truly cannot see why this is breaking.

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by BloodyIron** [comment]
To clarify when I say "the two pods get terminated and regenerate" I mean:

1. The NGINX Ingress and librespeed pods
2. They could regenerate on the same k8s node they were on previously, or any different k8s node, and the breakage happens regardless of whichever node they regenerate on.

I hope that the two failure scenarios are related by whatever the source of the problem is, but I really am stuck. Hopefully I can get some help with this soon :(

**Comment by BloodyIron** [comment]
Still really not finding why this is breaking. Would love to get some help troubleshooting this.

**Comment by BloodyIron** [comment]
Welp I guess back to trying other technologies instead...

**Comment by volatilemolotov** [comment]
Really not sure where to look for issues here. 

Can you post output of `cilium status` from any cilium pod. Perhaps we can see something there

**Comment by alphabet5** [comment]
What's the output of `iptables-save`?
What's the output of `ip a`?
Are all nodes masters?


A couple of guesses on what's happening without any cilium config, or other information.

- iptables rules remain in place after kube-proxy is disabled, so when pods move around, things break. (`iptables-save | grep -v KUBE | iptables-restore` would fix this, or rebooting nodes after disabling cilium.)

- you have multiple interfaces and cilium isn't running on some interfaces. (I've seen this with a main bonded interface, and other dedicated metallb interfaces.)

**Comment by BloodyIron** [comment]
Thanks for the attempts to help folks, but I've already moved on and just switched to Calico. I didn't have time to wait around trying to troubleshoot unreliable software that didn't log what the problem was (as a default action). Calico not only solves my instability problem, but also the SourceIP problem I originally aimed to solve. Wish I had better things to say, but this is the truth of what I experienced.
